lib/cuba.rb

```diff
@@ -86,11 +86,26 @@ def self.settings
   end
 
   def self.deepclone(obj)
-    Marshal.load(Marshal.dump(obj))
+    # Hashes with a default_proc cannot be serialized by Marshal.dump.
+    if obj.respond_to?(:default_proc)
```

Comment:
I took a clue from the argument name (`obj`). Because `Cuba.deepclone` is part of the public API, this patch could break code that calls `Cuba.deepclone` with anything other than a Hash.